packages/cli/src/commands/preview/server/livereload.ts

```diff
@@ -28,8 +28,17 @@ function requireParam(
   res: ServerResponse
 ) {
   try {
-    const url = new URL(req.url!, `http://${req.headers.host}`);
-    return JSON.parse(url.searchParams.get(param)!);
+    if (!req.url) {
+      throw new Error("No URL");
+    }
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    const paramFromSearch = url.searchParams.get(param);
+    if (!paramFromSearch) {
+      throw new Error(`Missing required param: ${param}`);
+    }
+
+    return JSON.parse(paramFromSearch);
```

Comment:
Doesn't this do the same thing with more code? Maybe we should just disable that rule or I'm missing something? Doesn't really seem like it helps in the test case either.

Reply:
🤷 I dunno. In my mind it's kind of like casting something as any/unknown, seems better to avoid if possible?

Reply:
oh ok, i had thought it was doing almost exactly the same thing with nicer syntax (asserting that it's not undefined)